Source/WebCore/rendering/RenderLayerBacking.cpp

```diff
@@ -1095,15 +1110,13 @@ bool RenderLayerBacking::updateConfiguration(const RenderLayer* compositingAnces
         updateContentsRects();
     }
 #endif
-#if ENABLE(WEBGL) || ENABLE(OFFSCREEN_CANVAS)
-    else if (renderer().isCanvas() && canvasCompositingStrategy(renderer()) == CanvasAsLayerContents) {
+    else if (shouldSetContentsDisplayDelegate()) {
         const HTMLCanvasElement* canvas = downcast<HTMLCanvasElement>(renderer().element());
```

Comment:
`auto`

Reply:
I'm kinda hesitant to fix up existing code that works unless necessary

Reply:
Guess I was wrong, the code surrounding it was using `auto` so this kinda sticks out. Changing to `auto*` then.